src/kernel/default.ts

```diff
@@ -655,7 +657,6 @@ class DefaultKernel implements Kernel.IKernel {
     if (this._token !== '') {
       url = url + `&token=${encodeURIComponent(this._token)}`;
```

Comment:
Is there always a guaranteed query string by this point in the logic? Because if not, this should probably be:
```typescript
url += `${url.search ? '&' : '?'}token=${encodeURIComponent(this._token)}`;
```